aten/src/ATen/native/README.md

```diff
@@ -384,6 +390,72 @@ will be automatically differentiated! This can be the case if the function imple
 only calls other operations which are themselves differentiable.  In this
 case, you don't have to write an entry in `tools/autograd/derivatives.yaml`.
 
+### Choosing the right dispatch keyword
+
+After writing a native function in C++, it's important to think about which dispatch keyword
+to use in native_functions.yaml as it gives dispatcher information about backend and autograd support
+of the implementation.
+
+Here're steps to follow to decide the right dispatch keyword:
+
+1. Think about inference, does your kernel work for all backends?
+
+    - No: you're likely providing different kerenls for different backends, e.g. DispatchStub or
+      backend-depenent logics are used in the implementation.
+      Write a dispatch section and enumerate all supported backends and point them to the implementations.
+      ```
+      dispatch:
+        CPU: kernel_cpu
+        CUDA: kernel_cuda
+        QuantizedCPU: kernel_quantized_cpu
+      ```
+
+      [END] Now this op will be called in `CPU/CUDA/QuantizedCPU` backend inference!
+
+      Note if you want this op to support training as well, you're required to write a formula in
+      derivatives.yaml since your backend implementations don't support autograd.
+
+    - Yes: you're likely calling other `at::` ops in the implemetation. Go to step 2.
+2. Think about training, does your kernel support autograd? [check autograd support](#will-your-function-be-automatically-differentiable)
+    - Yes: in other words, you're providing a `Math` kernel which supports both inference and autograd.
+      To use autograd support for training, simply skip adding a dispatch section or write
+      ```
+      dispatch:
+        Math: kernel
+      ```
+
+      [END] This will allow this op to be correctly registered for both inference and training.
+
+    - Yes, but you still want to provide a numerically stable gradient formula instead of using autorgrad, write
+      ```
+      dispatch:
+        DefaultBackend: kernel
+      ```
+
+      [END] Now this op is only used in inference. Note that now you're required to add a autograd formula,
+      or it'll error out in forward pass when calling with a Tensor has requires_grad=True.
+
+    - No: ops in this category are mainly `_out` functions that don't have a derivative formula defined.
```

Comment:
It might be worth explicitly adding the term "boilerplate" here. It's not so easy to see what exactly you mean by `_out` here; in particular, you are talking about a function that is calling an `_out` function, not an `_out` function itself.

Reply:
+1